r2-core/src/main/java/com/linkedin/r2/transport/common/AbstractClient.java

```diff
@@ -89,7 +89,13 @@ public void restRequest(RestRequest request, RequestContext requestContext, Call
     // This is needed as the legacy R2 server (before 2.8.0) does not support chunked transfer encoding.
     requestContext.putLocalAttr(R2Constants.IS_FULL_REQUEST, true);
     // here we add back the content-length header for the response because some client code depends on this header
-    streamRequest(streamRequest, requestContext, Messages.toStreamCallback(callback, true));
+
+    boolean addContentLengthHeader = true;
+    if ("HEAD".equalsIgnoreCase(request.getMethod())) {
+      addContentLengthHeader = false;
+    }
```

Comment:
Isn't there a way to get this as an enum? Feels like a string equality check is a little weak... Otherwise, this feels better anyway:
```java
boolean addContentLengthHeader = !"HEAD".equalsIgnoreCase(request.getMethod());
```